src/util/getDiff.ts

```diff
@@ -1,26 +1,68 @@
 import { assert } from "../util/assert";
+import { JimpClass } from "@jimp/types";
+import { Jimp } from "jimp";
+import { methods } from "@jimp/plugin-resize";
+import { clone } from "@jimp/utils";
 import pixelMatch from "pixelmatch";
-import { PNG } from "pngjs/browser";
 
-export function getDiff(aPng: PNG, bPng: PNG) {
-  assert(aPng.width === bPng.width && aPng.height === bPng.height);
-  const diff = new PNG({ width: aPng.width, height: bPng.height });
-  const diffPixelCount = pixelMatch(
-    aPng.data,
-    bPng.data,
-    diff.data,
-    aPng.width,
-    aPng.height,
+export function diff<I extends JimpClass>(img1: I, img2: I) {
+  let bmp1 = img1.bitmap;
+  let bmp2 = img2.bitmap;
+
+  if (bmp1.width !== bmp2.width || bmp1.height !== bmp2.height) {
+    if (bmp1.width * bmp1.height > bmp2.width * bmp2.height) {
+      // img1 is bigger
+      bmp1 = methods.resize(clone(img1), {
+        w: bmp2.width,
+        h: bmp2.height,
+      }).bitmap;
+    } else {
+      // img2 is bigger (or they are the same in area)
+      bmp2 = methods.resize(clone(img2), {
+        w: bmp1.width,
+        h: bmp1.height,
+      }).bitmap;
+    }
+  }
+
+  // eslint-disable-next-line @typescript-eslint/no-explicit-any
+  const diff = new (img1 as any).constructor({
```

Comment:
Is there a reason to not just use `new Jimp`?

Reply:
ok,i modify it.